spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/SparkTestHelperBase.java

```diff
@@ -55,12 +55,15 @@ private Object[] toJava(Row row) {
 
   protected void assertEquals(
       String context, List<Object[]> expectedRows, List<Object[]> actualRows) {
-    Assert.assertEquals(
-        context + ": number of results should match", expectedRows.size(), actualRows.size());
+    Assertions.assertThat(actualRows.size())
+        .as(context + ": number of results should match")
+        .isEqualTo(expectedRows.size());
     for (int row = 0; row < expectedRows.size(); row += 1) {
       Object[] expected = expectedRows.get(row);
       Object[] actual = actualRows.get(row);
-      Assert.assertEquals("Number of columns should match", expected.length, actual.length);
+      Assertions.assertThat(actual.length)
```

Comment:
for the same reason that I mentioned above, it's better to have `Assertions.assertThat(actual).as("Number of columns should match").hasSameSizeAs(expected);`